### PR TITLE
prevent changelist properties from being proxyed to underlying object. 

### PR DESCRIPTION
**Pull request by arenoir**
Quick fix for Issue #434

### CODE REVIEW

File: addon/index.js [hunk 1]
```diff
@@ -146,6 +158,10 @@ export class EmberChangeset extends BufferedChangeset {
     // 'person.username'
     let [baseKey, ...remaining] = key.split('.');
 
+    if (NOPROXYPROPS.includes(key)) {
+      return this[key];
+    }
+
```
Comment (snewcomer) [inline addon/index.js]: What do you think about this below on line 190?

```js
if (typeof this[key] !== 'undefined') {
```

Comment (arenoir) [reply addon/index.js]: There is a lot of complication here and off the top of my head wouldn't the type of the property `isValid` return [boolean](https://github.com/validated-changeset/validated-changeset/blob/master/src/index.ts#L209)?

Comment (snewcomer) [reply addon/index.js]: Yep and that is the goal.  The simple fix here is to change below from.

```js
if (this[key]) {
```

```js
if (typeof this[key] !== 'undefined') {
```

`isValid` and other properties that also exist on an Ember Data model won't be reflected via accessing through a changeset.    If the user asks the question `isValid`, the answer they should get back is if the changeset is valid.

**Comment by arenoir** [comment]
I’ll update the pull requests and test it on Monday.

**Comment by snewcomer** [comment]
@arenoir For context of a fix I added to `validated-changeset`

https://github.com/validated-changeset/validated-changeset/pull/18

File: addon/index.js (hunk 1 above)
Comment (arenoir) [reply addon/index.js]: I updated my pull request and rebased.

**Comment by ohcibi** [comment]
@snewcomer isnt the issue in that line https://github.com/poteto/ember-changeset/pull/439/commits/234f66ef23e6e327507ab574e8840ce09f5de0ad#diff-7562eb8eba3c3d86a671045f8c70107eL192 the same? If the test for `key` fails then the test for `baseKey` would fail in the same way as before?

**Pull request merged**